### PR TITLE
Limit SegmentWeight / SegmentDuration fewer bits

### PR DESCRIPTION
**Pull request by TheMarex**
# Issue

This PR explores #3896. As already described in the issue 16bit does not work because of ferries. 20bit seems fine (1747 hours per segment edge).

## Tasklist
 - [ ] ~~Use packed vector to get memory savings~~ Needs https://github.com/Project-OSRM/osrm-backend/issues/3952
 - [x] review
 - [x] adjust for comments

## Requirements / Relations

~~Requires https://github.com/Project-OSRM/osrm-backend/pull/3928~~

### CODE REVIEW

**Comment by daniel-j-h** [comment]
Does clipping trigger on the planet? Can you add a detailed warning in there if it happens.

**Comment by TheMarex** [comment]
> Does clipping trigger on the planet? Can you add a detailed warning in there if it happens.

This PR is primarily there to explore this. We should see a warning in the server logs. Unless there is a straight light ferry connection that takes 72 days (8735 km at 5km/h), I'm optimistic this will not cause problems.

File: include/util/typedefs.hpp [hunk 1]
```diff
@@ -58,6 +58,9 @@ using NodeID = std::uint32_t;
 using EdgeID = std::uint32_t;
 using NameID = std::uint32_t;
 using EdgeWeight = std::int32_t;
+using EdgeDuration = std::int32_t;
+using SegmentWeight = std::int32_t;
```
Comment (oxidase) [inline include/util/typedefs.hpp]: here and below can be used unsigned value. 

also unpacking should extend sign if type is signed. it is not used atm but better to add a static assert that the base type of the packed vector is unsigned

File: src/extractor/compressed_edge_container.cpp [hunk 2]
```diff
@@ -134,7 +135,21 @@ void CompressedEdgeContainer::CompressEdge(const EdgeID edge_id_1,
     // weight1 is the distance to the (currently) last coordinate in the bucket
     if (edge_bucket_list1.empty())
     {
-        edge_bucket_list1.emplace_back(OnewayCompressedEdge{via_node_id, weight1, duration1});
+        if (weight1 >= INVALID_SEGMENT_WEIGHT)
```
Comment (oxidase) [inline src/extractor/compressed_edge_container.cpp]: better `, clip(weight1), ` with `clip = [&clipped_weights] (EdgeWeight) -> SegmentWeight ` closure to reuse in both places

**Comment by TheMarex** [comment]
@oxidase could you take a look again if these changes work for you?

**Pull request merged**